orm/models.py

```diff
@@ -222,6 +222,13 @@ def limit(self, limit_count: int):
         )
         kwargs = validator.validate(kwargs)
 
+        # Remove fields that are allowed to be NULL or auto incremented
+        kwargs = {
+            key: value
+            for key, value in kwargs.items()
+            if not (fields[key].allow_null and value is None)
+        }
```

Comment:
Wouldn't it be better to do this only for primary keys, which is what is causing the issue in #21?

Something like:

```python
# Prevent inserting `NULL` as primary key.
if kwargs[self.model_cls.__pkname__] is None:
    kwargs.pop(self.model_cls.__pkname__)
```

Reply:
Hey.. I initially took similar approach, but I ran into other issue:
- nullable fields being inserted with 'null' (I guess this is happening because the typesystem sets to None after validation, haven't had the time to investigate, I actually need to check this on the update statement, I think that depending on how sqlalchemy introspects the statement this could also happend there)
It's also less data going through, what do you think?

Reply:
> typesystem sets to None after validation

Yes, exactly.

I investigated, and it seems the issue we're having with the primary key is that `orm` sets `allow_null=True` if `primary_key=True` (see [here](https://github.com/encode/orm/blob/master/orm/fields.py#L15)), which TypeSystem picks up to set `default=None` (see [here](https://github.com/encode/typesystem/blob/master/typesystem/fields.py#L35)).

This won't happen if the field is non-PK: TypeSystem will treat the field as required, and require that a non-`None` value is passed.

Example model with a non-nullable field:

```python
class Post(orm.Model):
    ...
    slug = orm.String(max_length=80)
```

Not passing a `slug` results in:

```python
{'slug': 'This field is required.'}
```

passing `slug=None` results in:

```python
{'slug': 'May not be null.'}
```

So my conclusion is that this issue only affects primary key fields. Or am I missing something? 😄

Reply:
Yep, it works perfectly with required fields. 

Actually my use case was a JSON field, I'll check this properly :)

Reply:
Solution you pointed works fine ;)

JSON null value is apparently the default for sqlalchemy. Not a big deal though.